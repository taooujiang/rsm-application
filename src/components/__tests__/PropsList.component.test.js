'use strict';

import React from 'react';
import PropsList from '../PropsList.component.js'
import renderer from 'react-test-renderer';
import {shallow,mount} from 'enzyme';

it('PropList component', () => {
  const tree = renderer
    .create(<PropsList labelPosition={"left"} labelWidth="120" labelSuffix={":"} inline={false}/>)
    .toJSON();
  expect(tree.type).toBe('ul')
  expect(tree).toMatchSnapshot();
});
it('render with PropListItem', () => {
    const tree = mount(<PropsList.Item label="姓名">一个人玩</PropsList.Item>)
    console.log(tree.name())
  //  expect(tree.type()).toBe("li")
    expect(tree.name()).toBe("PropsListItem")
    expect(tree.props().label).toBe("姓名")
    //expect(tree.type).toBe('li')
})
 
it('render with children', () => {
  const tree = renderer.create(<PropsList labelPosition={"left"} labelWidth="120" labelSuffix={":"} inline={false}>
    <PropsList.Item label="姓名">
     一个人玩
    </PropsList.Item>
    <PropsList.Item label="性别">
     男
    </PropsList.Item>
  </PropsList>).toJSON();


  expect(tree.children.length).toBe(2)

  expect(tree).toMatchSnapshot();
});

it('render with children shallow',()=>{

  const propList = mount(
    <PropsList labelPosition={"left"} labelWidth="120" labelSuffix={":"} inline={false}>
      <PropsList.Item label="姓名">
       一个人玩
      </PropsList.Item>
      <PropsList.Item label="性别">
       男
      </PropsList.Item>
    </PropsList>
  );
  expect(propList.text()).toEqual('姓名:一个人玩性别:男');
})
it('render with dynamic items', () => {
  var items=[{
    label:"姓名",
    value:"一个人玩"
  },{
    label:"性别",
    value:"男"
  },{
    label:"age",
    value:"19"
  }]
  const tree = renderer
    .create(
    <PropsList items={items} labelPosition={"left"} labelWidth="120" labelSuffix={":"} inline={false}>
    </PropsList>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
