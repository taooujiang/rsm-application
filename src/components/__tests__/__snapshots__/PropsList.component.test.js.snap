// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropList component 1`] = `
<ul
  className="props-list"
/>
`;

exports[`render with children 1`] = `
<ul
  className="props-list"
>
  <li
    className="props-item"
  >
    <label
      className="props-label"
    >
      姓名
      :
    </label>
    一个人玩
  </li>
  <li
    className="props-item"
  >
    <label
      className="props-label"
    >
      性别
      :
    </label>
    男
  </li>
</ul>
`;

exports[`render with dynamic items 1`] = `
<ul
  className="props-list"
>
  <li
    className="props-item"
  >
    <label
      className="props-label"
    >
      姓名
      :
    </label>
    一个人玩
  </li>
  <li
    className="props-item"
  >
    <label
      className="props-label"
    >
      性别
      :
    </label>
    男
  </li>
  <li
    className="props-item"
  >
    <label
      className="props-label"
    >
      age
      :
    </label>
    19
  </li>
</ul>
`;
